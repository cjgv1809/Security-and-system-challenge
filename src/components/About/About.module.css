.about {
  padding: 1.5rem;
  background: var(--color-background-light);
  border-radius: 16px;
  box-shadow: 0 4px 30px var(--color-box-shadow);
  backdrop-filter: blur(6px);
  -webkit-backdrop-filter: blur(6px);
  border: 1px solid var(--color-border-light);
  width: min(100%, 600px);
  margin: 0 auto;
}

.title {
  font-size: 1.5rem;
  margin-bottom: 0.625rem;
}

.description {
  font-size: 1rem;
  color: var(--color-item-text);
  line-height: 1.6;
}
