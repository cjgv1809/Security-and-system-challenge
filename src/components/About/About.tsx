import React from "react";
import styles from "./About.module.css";

type AboutProps = {
  description: string;
};

const About: React.FC<AboutProps> = ({ description }) => {
  return (
    <section className={styles.about}>
      <h2 className={styles.title}>Sobre m√≠</h2>
      <p className={styles.description}>{description}</p>
    </section>
  );
};

export default About;
