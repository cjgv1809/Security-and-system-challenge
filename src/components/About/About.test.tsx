import React from "react";
import { render, screen } from "@testing-library/react";
import About from "./About";

describe("About", () => {
  it("renders the about section", () => {
    const description = "Desarrollador de software con 5 años de experiencia";
    render(<About description={description} />);

    expect(
      screen.getByRole("heading", { name: /sobre mí/i })
    ).toBeInTheDocument();
    expect(screen.getByText(description)).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<About description="Sobre Joe" />);

    expect(container).toMatchSnapshot();
  });
});
