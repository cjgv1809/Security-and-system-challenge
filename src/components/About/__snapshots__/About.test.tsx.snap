// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About matches snapshot 1`] = `
<div>
  <section
    class="about"
  >
    <h2
      class="title"
    >
      Sobre m√≠
    </h2>
    <p
      class="description"
    >
      Sobre Joe
    </p>
  </section>
</div>
`;
